### ISSUE TITLE
🐛 BUG: Remote markdown inside of a .map causes a parse error

### ISSUE THREAD

**Issue by mattjennings** [snowpackjs/astro#1060]
### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

When using the Markdown component with the `content` prop inside of a .map expression, astro throws a parse error:

```
[parse-error]

 /sandbox/src/pages/index.astro:16:13

 𝘅 Expected ")" but found end of file

     14 |     {data.map(c => (
     15 |       <Markdown content={c} />
   > 16 |     ))}
        |        ^
     17 |   </body>
     18 | </html>

(node:144) UnhandledPromiseRejectionWarning: Error: Unable to compile expression
    at compileExpression (file:///sandbox/node_modules/astro/src/compiler/codegen/index.ts:256:28)
    at Object.enter (file:///sandbox/node_modules/astro/src/compiler/codegen/index.ts:606:24)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:49:5)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:86:6)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at AsyncWalker.visit (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/async.js:79:14)
    at asyncWalk (file:///sandbox/node_modules/astro/node_modules/estree-walker/src/index.js:34:9)
```

### Steps to Reproduce

Create a .astro file with the following content:

```astro
---
import { Markdown } from 'astro/components';

const content = "## Hello";
const data = [content, content];

---

<html>
  <head>
    <title>Hello Astro!</title>
  </head>
  <body>
    {data.map(c => (
      <Markdown content={c} />
    ))}
  </body>
</html>
```

### Link to Minimal Reproducible Example (Optional)

https://codesandbox.io/s/stoic-perlman-khgy9?file=/src/pages/index.astro

**Comment by mattjennings** [comment]
Seems to work if Markdown is wrapped by an outer element.

```jsx
    {data.map(c => (
      <div>
        <Markdown content={c} />
      </div>
    ))}
```

**Comment by natemoo-re** [comment]
Confirming that this is finally fixed in `astro@next`! https://codesandbox.io/s/awesome-hill-lu44v?file=/src/pages/index.astro

---

The team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
